test_conformance/commonfns/test_binary_fn.c

```diff
@@ -141,7 +141,7 @@ int test_binary_fn( cl_device_id device, cl_context context, cl_command_queue qu
         }
         const char *ptr = programSrc;
         err = create_single_kernel_helper( context, &program[ i ], &kernel[ i ], 1, &ptr, "test_fn" );
-        test_error( err, "Unable to create kernel" );
+        test_failure_warning(CL_SUCCESS, err, "Unable to create kernel" );
```

Comment:
Using `test_failure_warning` is confusing. Maybe introducing a new error checking macro to deal with this is justified. What you want isn't so much warnings but deferred errors.

Reply:
sure, and agreed